src/useMutationObserver.ts

```diff
@@ -0,0 +1,69 @@
+import useEffect from './useIsomorphicEffect'
+
+/**
+ * Efficiently observe size changes on an element. Depends on the `ResizeObserver` api,
+ * and polyfills are needed in older browsers.
+ *
+ * ```ts
+ * const [element, attachRef] = useCallbackRef(null);
+ *
+ * useMutationObserver(element, { subtree: true }, (records) => {
+ *
+ * });
+ *
+ * return (
+ *   <div ref={attachRef} />
+ * )
+ * ```
+ *
+ * @param element The DOM element to observe
+ * @param config The observer configuration
+ * @param callback A callback fired when a mutation occurs
+ */
+function useMutationObserver(
+  element: Element | null | undefined,
+  config: MutationObserverInit,
+  callback: MutationCallback,
+): void {
+  const {
+    attributeFilter,
+    attributeOldValue,
+    attributes,
+    characterData,
+    characterDataOldValue,
+    childList,
+    subtree,
+  } = config
```

Comment:
is there really no good way to do shallow equality here? seems awkward to destructure config just to pass it back in again "re-structed", as it were

Reply:
nope :/ could wrap it into a hook but it adds an extra layer of execution that seems worse

Reply:
i feel like it'd be a useful hook! we'd use it in react-relay-mutation at least...

Reply:
added!